include/Utilities.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <iostream>
+#include <vector>
+#include <memory>
+#include <string>
+#include <map>
+
+using std::cout;
+using std::cin;
+using std::endl;
+using std::string;
+using std::unique_ptr;
+using std::vector;
+using std::make_unique;
+using std::move;
+
+class Piece;
+
+using board_vec = vector<vector<unique_ptr<Piece>>>;
```

Comment:
Why are the pieces in the board `unique_ptr`s? Would it be enough to store them directly without the pointer indirection?